app/models/lookup_keys/variable_lookup_key.rb

```diff
@@ -16,8 +16,12 @@ def param_class
     puppetclass
   end
 
-  def self.humanize_class_name
-    "Smart variable"
+  def self.humanize_class_name(options = nil)
```

Comment:
After changing the validation the error added tries to call `VariableLookupKey.human_attribute_name(:puppetclass_id)` to use in the error message. This got to `humanize_class_name` and failed because it doesn't expect any attributes.